.github/workflows/docker-publish.yml

```diff
@@ -1,82 +1,89 @@
-# Source: https://raw.githubusercontent.com/foundry-rs/foundry/master/.github/workflows/docker-publish.yml
-name: docker
+#
+name: Create and publish a Docker image
 
 on:
   workflow_dispatch:
     inputs:
       light-client:
         description: 'aptos or ethereum'
         type: choice
+        required: true
         options:
           - aptos
           - ethereum
+      version:
+        description: "Image version"
+        required: true
+        type: string
 
+permissions:
+  id-token: write
+  contents: read
+  attestations: write
+  packages: write
+
+# Defines two custom environment variables for the workflow. These are used for the Container registry domain, and a name for the Docker image that this workflow builds.
 env:
-  REGISTRY: ghcr.io
+  REGISTRY: "ghcr.io"
+  NAMESPACE: "argumentcomputer"
+  IMAGE_NAME: "proof-server"
+# There is a single job in this workflow. It's configured to run on the latest available version of Ubuntu.
 jobs:
-  container:
+  build-and-push-image:
     runs-on: ubuntu-latest
-    # https://docs.github.com/en/actions/reference/authentication-in-a-workflow
+    # Sets the permissions granted to the `GITHUB_TOKEN` for the actions in this job.
     permissions:
-      id-token: write
-      packages: write
       contents: read
-    timeout-minutes: 120
+      packages: write
+      attestations: write
+      id-token: write
+      #
     steps:
       - name: Checkout repository
-        id: checkout
         uses: actions/checkout@v4
-
-      - name: Install Docker BuildX
-        uses: docker/setup-buildx-action@v2
-        id: buildx
-        with:
-          install: true
-
-      # Login against a Docker registry except on PR
-      # https://github.com/docker/login-action
-      - name: Log into registry ${{ env.REGISTRY }}
-        # Ensure this doesn't trigger on PR's
-        if: github.event_name != 'pull_request'
-        uses: docker/login-action@v2
+      # Uses the `docker/login-action` action to log in to the Container registry registry using the account and password that will publish the packages. Once published, the packages are scoped to the account defined here.
```

Comment:
From https://docs.docker.com/build/ci/github-actions/

I would introduce before this:
```
      # Set up Docker Buildx for better layer caching
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2

      # Use Docker layer caching for faster builds
      - name: Cache Docker layers
        uses: actions/cache@v3
        with:
          path: /tmp/.buildx-cache
          key: ${{ runner.os }}-buildx-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-buildx-
```